rclip/fs.py

```diff
@@ -31,16 +30,9 @@ def walk(
   while dirs_to_process:
     dir = dirs_to_process.pop()
     with os.scandir(dir) as it:
-      base_names = {}
-      sorted_entries = sorted(it, key=(
-        lambda entry: os.path.splitext(entry.name)[1].lower() not in PRIORITIZED_IMAGE_EXTENSIONS
-      ))
```

Comment:
@abidkhan484, this is super slow and possibly buggy because it attempts to list the whole file list on every file. We should've never done this.

Reply:
Actually I tried with less files, that's why I couldn't find it. But I got your point.

Previously, a generator was returned in the implementation and in this implementation, the conversation is the bottleneck.

Could you share any idea on how the prioritized extensions can be settled?

Reply:
@abidkhan484, the generator is the solution for the bottleneck.

Reply:
@abidkhan484, check out this implementation: https://github.com/yurijmikhalevich/rclip/pull/140/commits/12608f54634fd48dccf53517d88826e354511ee1

With this implementation, we keep the generator to list files, maintaining the low RAM profile. The only change is that now, if the file is RAW, we check whether there is a supported processed file alongside it. This check will take some extra time, but it's fine because the time amount is predictable, constant, and only affects users that have RAW files to enable that new feature for them (the cost of enabling the feature).

The original implementation was a huge downgrade for all rclip users, even for those who never searched RAW files.